src/gui/dialogs/unit_advance.hpp

```diff
@@ -25,7 +25,7 @@ class unit_advance : public modal_dialog
 public:
 	unit_advance(const std::vector<unit_const_ptr>& samples, std::size_t real);
 
-	int get_selected_index() const
+	std::size_t get_selected_index() const
```

Comment:
The only caller of this casts it back to `int`, so this seems a bad change. Both the caller and the old method signature knew something that `std::size_t` doesn't match, so the question should be, is `selected_index_` the wrong type?

I think it is the wrong type, it's being set to the value of `int listbox::get_selected_row()`.

Reply:
If the caller wants it as int for its own reasons, then it's free to do that. It seems kind of weird to have it return a signed value when it can actually only ever return unsigned values though.

Reply:
Updated.